tests/test_ac_harness_auto_drive.py

```
@@ -0,0 +1,239 @@
+    assert report.ok is False
+    assert report.stage == "pipeline"
+    assert report.error is not None and "sidecar exploded" in report.error
+    assert record["stop_set"] is True  # teardown still ran
+    assert ctrl.closed is True
+
+
+def test_skip_launch_does_not_call_launch():
+    report = asyncio.run(
+        run_auto_drive(
+            _cfg(skip_launch=True),
+            launch=lambda c: pytest.fail("launch must be skipped"),
```

Ruff (0.15.18)

[warning] Unused lambda argument: `c`

(ARG005)